app/javascript/components/rooms/room/join/JoinCard.jsx

```diff
@@ -61,6 +61,19 @@ export default function JoinCard() {
 
   const path = encodeURIComponent(document.location.pathname);
 
+  // get queryParams for JoinFormName
+  const location = useLocation();
+  const queryParams = new URLSearchParams(location.search);
+  const joinFormName = queryParams.get('joinFormName');
+
+  useEffect(() => { // Default join name input to the value of joinFormName if it's provided in the query parameters
+    if (joinFormName) {
+      methods.setValue('name', joinFormName);
+    } else if (currentUser?.name) {
+      methods.setValue('name', currentUser.name);
+    }
+  }, [joinFormName, currentUser?.name]);
+
```

Comment:
Isn't there already another useEffect that does this?

Reply:
yes I see it I'll modify that one instead and update